### ISSUE TITLE
subAccount provider that depend on mainAccount?

### ISSUE THREAD

**Issue by nicolasburtey** [RJPearson94/terraform-provider-twilio#70]
I'm trying to create a twilio verify from a subAccount. the accountId and Authtoken I initially have are from the main account. So I need to create 2 twilio providers, one for the main account, and one for the sub account.

but the twilio provider would not let me create the sub account with a dependancy on the main account. The error that is returned is: `Error: Account details specified are invalid. Validation errors: [SID is required, Auth token is required]`

here is the code that would reproduce the issue: 

```
terraform {
  required_providers {
    twilio = {
      source  = "RJPearson94/twilio"
      version = ">= 0.2.1"
    }
  }
}

locals {
  twilio_account_sid = "AC_Main_Account"
  twilio_auth_token  = "token"
  name_prefix = "terraform-test"
}

provider "twilio" {
  alias       = "main"
  account_sid = local.twilio_account_sid
  auth_token  = local.twilio_auth_token
}

resource "twilio_account_sub_account" "galoy_sub_account" {
  provider      = twilio.main
  friendly_name = "${local.name_prefix}-sub-account"
}

provider "twilio" {
  alias       = "sub"
  account_sid = twilio_account_sub_account.galoy_sub_account.sid
  auth_token  = twilio_account_sub_account.galoy_sub_account.auth_token
}

resource "twilio_verify_service" "service" {
  provider       = twilio.sub
  friendly_name  = local.name_prefix
  lookup_enabled = false
}

output "result" {
  value = twilio_verify_service.service.id
}
```

**Comment by RJPearson94** [comment]
Hi @nicolasburtey,

Thanks for using the provider and apologies for the issue. The Terraform provider does not natively support this use case as the validation is present to ensure that the provider has the necessary config to manage resources in Twilio.

If you want to support this behaviour I would recommend splitting the creation of the sub-account and sub-account resources into separate states. 

So you would create your sub-account first, then configure a separate provider with the necessary credentials to provision sub-account resources. You can use [Terragrunt](https://terragrunt.gruntwork.io/) to manage the dependencies between both sets of resources.  

I have created an example and documentation to demonstrate this. The code and docs can be found at https://github.com/RJPearson94/terraform-provider-twilio/tree/main/examples/account/sub-account. 

I know this isn't the answer you were looking for but I hope it helps.
If this is not suitable, then please let me know. As it has been raised by another member of the community, their workaround can be found at https://github.com/RJPearson94/terraform-provider-twilio/issues/54

Kind Regards
Rob

**Comment by nicolasburtey** [comment]
hi @RJPearson94 thanks for your quick answer!

adding Terragrunt dependancy doesn't sound ideal. it's another tool and comes with added complexity.

I'm going to try the hack from https://github.com/RJPearson94/terraform-provider-twilio/issues/54 for now. 

Here is maybe a suggestion: add an argument `nocheck` to the provider, that when it's `true` would not check the presence of an existing AccountId / Auth string. This would allow this use case? I'm not an expert in terraform so maybe I'm missing something

**Comment by RJPearson94** [comment]
hi @nicolasburtey,

ok, I have done some investigation and your idea does appear to work. So I have added a new attribute to the provider to allow you to skip credential validation. An example can be seen below

```hcl
...

provider "twilio" {
  alias       = "sub"
  skip_credential_validation = true
  account_sid = twilio_account_sub_account.sub_account.sid
  auth_token  = twilio_account_sub_account.sub_account.auth_token
}

...
```

You may run into issues if the sub-account is tainted but it appears to work as expected.

This capability has been included in v0.22.0, which is now available in the Terraform Registry.

Please update your Terraform configuration to include the following

```hcl
terraform {
  required_providers {
    twilio = {
      source = "RJPearson94/twilio"
      version = ">= 0.22.0"
    }
  }
} 
```

Hope this helps

Kind Regards
Rob

**Comment by nicolasburtey** [comment]
awesome. trying it now.
